concepts/mesh_with_materials/python/material.py

```diff
@@ -57,11 +57,11 @@ def attach_texture(
         os.path.join(ASSETS_DIRECTORY, "USDLogoLrg.png")
     )
     diffuseTextureSampler.CreateInput("st", Sdf.ValueTypeNames.Float2).ConnectToSource(
-        reader, "result"
+        reader.CreateInput("result", Sdf.ValueTypeNames.Float2)
     )
     diffuseTextureSampler.CreateOutput("rgb", Sdf.ValueTypeNames.Float3)
     shader.CreateInput("diffuseColor", Sdf.ValueTypeNames.Color3f).ConnectToSource(
-        diffuseTextureSampler, "rgb"
+        diffuseTextureSampler.CreateInput("rgb", Sdf.ValueTypeNames.Color3f)
```

Comment:
Here are some more guesses on input vs output and type.   The examples all run correctly, just not sure if they're doing the right thing.